Remove MacOS build from CircleCI, use Github Actions (through Gitlab) instead

albertvaka opened the pull request:
### What does this PR do?

CircleCI hits the OpenSSL 1.0.2 & Let's Encrypt issue when using the latest build scripts from [datadog-agent-macos-build](https://github.com/DataDog/datadog-agent-macos-build) before installing Python 3.8.11.

This was working until now because we were still installing Python 3.8.5, which is another reason to remove this from CircleCI: Gitlab just triggers Github actions which should always run up-to-date scripts.

Depends on: https://github.com/DataDog/datadog-agent-macos-build/pull/76

albertvaka committed:
Update repo that contains MacOS build scripts

mx-psi reviewed (dismissed):
Looks good! After this we also need to update the 'datadog-agent-macos-build repository' section on the doc for upgrading the Go version on mars jenkins

albertvaka committed:
Move MacOS tests from CircleCI to Gitlab

CircleCI hits the OpenSSL 1.0.2 & Let's Encrypt isue when installing
Python 3.8.11. This worked before because we were still installing 3.8.5,
which is another reason to remove this from CircleCI: Gitlab just triggers
Github actions which should always run up-to-date scripts.

albertvaka committed:
Trigger MacOS unit tests on Github Actions

albertvaka committed:
Use different workflows instead of passing a variable

albertvaka committed:
Remove test code

albertvaka committed:
Fail on any error, not just 401

albertvaka committed:
Remove some of the default inputs since test.yaml won't accept them

albertvaka committed:
Pass workflow to get_macos_workflow_run_for_ref
